### ISSUE TITLE
Problem convert with blocks and retextured default items

### ISSUE THREAD

**Issue by PanHaskins** [BrilliantTeam/Minecraft-ResourcePack-Migrator#1]
When converting, custom blocks go through just fine but basic minecraft blocks look like they are textureless + it has a problem with re-textured basic items like Spawn Egg.
![image](https://github.com/user-attachments/assets/a4cc4bb2-4804-4108-b6ad-a7b3f1fc1d9d)
![image](https://github.com/user-attachments/assets/9093ced7-f8b7-43a4-a61c-5115b7df7fe1)
![image](https://github.com/user-attachments/assets/c965422f-e002-4c5f-b826-e73362a34b0d)

Client Log: [https://pastebin.com/EsG8awqm](https://pastebin.com/EsG8awqm)
Converting from 1.20.4 -> 1.21.4

**Comment by RICE0707** [comment]
> When converting, custom blocks go through just fine but basic minecraft blocks look like they are textureless + it has a problem with re-textured basic items like Spawn Egg. ![image](https://private-user-images.githubusercontent.com/58113388/393536272-a4cc4bb2-4804-4108-b6ad-a7b3f1fc1d9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MDMwNjksIm5iZiI6MTczMzgwMjc2OSwicGF0aCI6Ii81ODExMzM4OC8zOTM1MzYyNzItYTRjYzRiYjItNDgwNC00MTA4LWI2YWQtYTdiM2YxZmMxZDlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEwVDAzNTI0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhNTNkZmQ0ZjI3MDUzN2MzZWMzNjQzMDZiNTE1NTkwMDRmYjhjMDZlYmQxZmJlZDE4N2E3YzExMGY4OTAwYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uB_8eJYLlmDUAsp5NX37nxL3DtCOf40M41L0fpZVxd8) ![image](https://private-user-images.githubusercontent.com/58113388/393536774-9093ced7-f8b7-43a4-a61c-5115b7df7fe1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MDMwNjksIm5iZiI6MTczMzgwMjc2OSwicGF0aCI6Ii81ODExMzM4OC8zOTM1MzY3NzQtOTA5M2NlZDctZjhiNy00M2E0LWE2MWMtNTExNWI3ZGY3ZmUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEwVDAzNTI0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2YWQ5MDJlMGUwNjRmZWM1ZjhkMDFhNjA1OGM3NDhmM2U0YzkzOGU1OThkNmFkNTE3Njg3ZGZlMjNkOTVhYTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Kyum2tlU1JLHrMnSpRE4S3Mp5Epj9oW8c_E7OILCUyU) ![image](https://private-user-images.githubusercontent.com/58113388/393539052-c965422f-e002-4c5f-b826-e73362a34b0d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MDMwNjksIm5iZiI6MTczMzgwMjc2OSwicGF0aCI6Ii81ODExMzM4OC8zOTM1MzkwNTItYzk2NTQyMmYtZTAwMi00YzVmLWI4MjYtZTczMzYyYTM0YjBkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEwVDAzNTI0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjMTMwNGU3MmU3YjUxYzkxYzliNTY1NjQxZjRhYzIwMGY5ZDA5YzJiYWRhNzU5MGY2MmUwOWY0YTdmNWE0YmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DO_PvpbnBnePLTQDqEUOg8AKRzbzFRanV06EvqvkALE)
> 
> Client Log: https://pastebin.com/EsG8awqm Converting from 1.20.4 -> 1.21.4

Just released version 1.2, which includes handling for the `minecraft:block/*` path.  

Additionally, this tool only supports conversions for older formats like:  
```
{
    "parent": "item/handheld",
    "textures": {
        "layer0": "item/stick"
    },
    "overrides": [
        {"predicate": {"custom_model_data": 19002}, "model":"custom_items/cat_hat/cat_hat_black"}
    ]
}
```  
If your resource pack contains entries with a format like:  
```
{
    "parent": "builtin/entity",
    "textures": {
        "particle": "block/soul_sand"
    },
    "gui_light": "front",
    "display": {
        ... omitted below 
```  
then this tool will not be able to assist you.

**Comment by RICE0707** [comment]
In the next version, I plan to attempt fixing issues related to block types.

**Comment by RICE0707** [comment]
For **`spawn_egg`**, my converter does not plan to apply any special handling.  

This is because in **1.21.4**, `spawn_egg` introduced a new `tints` parameter, where the `value` differs for each `spawn_egg`.  

I don't intend to define the `value` for every single `spawn_egg`, as it would be too tedious.

**Comment by RICE0707** [comment]
Regarding shields, since I don't know how to create them between versions 1.14 and 1.21.3, I am unable to write code to convert them to 1.21.4. If someone can provide me with a clear and complete example resource pack, I might be able to implement the conversion for shields.

In the recently released version 1.2.4, I addressed the handling of `"parent": "minecraft:block/*"`, which should fix many issues with vanilla block textures after conversion.

Additionally, the following items have received special handling: chest, trapped_chest, potion, splash_potion, lingering_potion.

**Comment by RICE0707** [comment]
Version 1.3.2 should solve the vanilla shield issue.
